### PR TITLE
fix(kensetsu): update cdp on liquidation and creation

### PR DESCRIPTION
**Pull request by Alexey-N-Chernyshov**

### CODE REVIEW

File: pallets/kensetsu/src/lib.rs [hunk 1]
```diff
@@ -1685,6 +1667,18 @@ pub mod pallet {
...
+                Error::<T>::WrongAssetId
+            );
+            ensure!(
+                *collateral_asset_id != T::KenAssetId::get(),
+                Error::<T>::WrongAssetId
+            );
+            ensure!(
+                *collateral_asset_id != T::KenAssetId::get(),
+                Error::<T>::WrongAssetId
+            );
```
Comment (wer1st) [inline pallets/kensetsu/src/lib.rs:1681]: Duplicate

**Review by wer1st — APPROVED** [review]

File: pallets/kensetsu/src/migrations.rs [hunk 2]
```diff
@@ -97,3 +97,66 @@ pub mod init {
...
+
+            for (_, cdp) in CDPDepository::<T>::iter() {
+                *collateral_debts.entry(cdp.collateral_asset_id).or_default() += cdp.debt;
+                total_debt += cdp.debt;
+                weight += <T as frame_system::Config>::DbWeight::get().reads(1);
+            }
+
+            for asset_id in CollateralInfos::<T>::iter_keys() {
+                CollateralInfos::<T>::mutate(asset_id, |collateral_info| {
+                    let collateral_info = collateral_info.as_mut().unwrap();
```
Comment (vovac12) [inline pallets/kensetsu/src/migrations.rs:134]: Please, avoid using `unwrap`s in migrations

Comment (Alexey-N-Chernyshov) [reply pallets/kensetsu/src/migrations.rs:134]: `asset_id` is from iter, entry must exists.

File: pallets/kensetsu/src/migrations.rs [hunk 3]
```diff
@@ -97,3 +97,66 @@ pub mod init {
...
+                        *collateral_debts.get(&asset_id).unwrap_or(&Balance::zero());
+                });
+                weight += <T as frame_system::Config>::DbWeight::get().writes(1);
+            }
+
+            // burn KUSD on tech account
+            let treasury_account_id = technical::Pallet::<T>::tech_account_id_to_account_id(
+                &T::TreasuryTechAccount::get(),
+            )
+            .unwrap();
```
Comment (Alexey-N-Chernyshov) [inline pallets/kensetsu/src/migrations.rs:145]: Tech account must exists on stage.

File: pallets/kensetsu/src/migrations.rs [hunk 4]
```diff
@@ -97,3 +97,66 @@ pub mod init {
...
+            let balance =
+                T::AssetInfoProvider::free_balance(&T::KusdAssetId::get(), &treasury_account_id)
+                    .unwrap();
```
Comment (Alexey-N-Chernyshov) [inline pallets/kensetsu/src/migrations.rs:148]: Tech account and kusd exists on stage.

File: pallets/kensetsu/src/migrations.rs [hunk 5]
```diff
@@ -97,3 +97,66 @@ pub mod init {
...
+            let to_burn = balance - total_debt;
+            assets::Pallet::<T>::burn_from(
+                &T::KusdAssetId::get(),
+                &treasury_account_id,
+                &treasury_account_id,
+                to_burn,
+            )
+            .unwrap();
```
Comment (Alexey-N-Chernyshov) [inline pallets/kensetsu/src/migrations.rs:156]: And this difference we want to burn is on tech account.

**Review by vovac12 — APPROVED** [review]

**Pull request merged**